src/composables/transaction.ts

```diff
@@ -22,33 +25,74 @@ interface ITokenTransactionComposable extends ITx {
   isAe: string
 }
 
-export const useTransactionToken = (
-  initTransaction: ITransaction | undefined,
-  showDetailedAllowanceInfo?: boolean,
-) => {
-  const transaction = ref<ITransaction | undefined>(initTransaction);
+interface UseTransactionOptions {
+  showDetailedAllowanceInfo?: boolean
+}
+
+export const useTransaction = ({
+  showDetailedAllowanceInfo,
+}: UseTransactionOptions = {}) => {
+  const transaction = ref<ITransaction | undefined>();
 
   function setTransaction(newTransaction: ITransaction) {
     transaction.value = newTransaction;
   }
 
-  const fungibleTokens = useState('fungibleTokens');
-  const availableTokens = computed<ITokenList>(() => fungibleTokens.value.availableTokens);
+  const availableTokens = useState('fungibleTokens', 'availableTokens');
 
-  const getTxAmountTotal = useGetter('getTxAmountTotal');
-  const isTxAex9 = useGetter('isTxAex9');
+  const account = useGetter<IAccount>('account');
+  const accounts = useGetter<IAccount[]>('accounts');
 
+  const isTxAex9 = useGetter('isTxAex9');
+  const getTxAmountTotal = useGetter('getTxAmountTotal');
   const getTxSymbol = useGetter('getTxSymbol');
   const getTxType = useGetter('getTxType');
   const getTxDirection = useGetter('getTxDirection');
   const getDexContracts = useGetter<IDexContracts>('getDexContracts');
+  const getExplorerPath = useGetter('getExplorerPath');
 
   const txType = computed<TransactionType>(() => getTxType.value(transaction.value));
 
   const isAllowance = computed(() => transaction.value
     && FUNCTION_TYPE_DEX.allowance.includes(transaction.value.tx.function)
     && availableTokens.value[transaction.value.tx.contractId]);
 
+  const ownershipStatus = computed(
+    () => {
+      const txAddress = transaction.value?.tx.accountId
+          || transaction.value?.tx.callerId;
+      if (account.value?.address === txAddress) {
+        return TRANSACTION_OWNERSHIP_STATUS.current;
+      }
+      if (accounts.value?.find((acc) => acc.address === txAddress)) {
+        return TRANSACTION_OWNERSHIP_STATUS.subAccount;
+      }
+      return TRANSACTION_OWNERSHIP_STATUS.other;
+    },
+  );
+
+  const ownershipAccount = computed(() => {
```

Comment:
Please add an interface for the return type. Something like a `IAccountExtended` / `IAccountLabeled`...